packages/internals/src/get-generators/generatorResolvers/prisma-client-js/prismaClientResolver.ts

```diff
@@ -30,27 +31,28 @@ export async function prismaClientResolver(baseDir: string, version?: string) {
   await checkTypeScriptVersion()
 
   if (!prismaClientDir && !process.env.PRISMA_GENERATE_SKIP_AUTOINSTALL) {
-    // TODO: should this be relative to `baseDir` rather than `process.cwd()`?
-    if (
-      !fs.existsSync(path.join(process.cwd(), 'package.json')) &&
-      !fs.existsSync(path.join(process.cwd(), '../package.json'))
-    ) {
-      // Create default package.json
-      const defaultPackageJson = `{
-"name": "my-prisma-project",
-"version": "1.0.0",
-"description": "",
-"main": "index.js",
-"scripts": {
-  "test": "echo \\"Error: no test specified\\" && exit 1"
-},
-"keywords": [],
-"author": "",
-"license": "ISC"
-}
-`
-      fs.writeFileSync(path.join(process.cwd(), 'package.json'), defaultPackageJson)
-      console.info(`✔ Created ${bold(green('./package.json'))}`)
+    let projectRoot = longestCommonPathPrefix(baseDir, process.cwd())
+    debug('projectRoot', projectRoot)
+
+    if (projectRoot === undefined) {
+      console.warn(
+        yellow(
+          `${bold('Warning:')} schema directory ${bold(baseDir)} and current working directory ${bold(
+            process.cwd(),
+          )} have no common ancestor, using the schema path as the project root.`,
+        ),
+      )
+      projectRoot = baseDir
+    }
+
+    if (!fs.existsSync(path.join(projectRoot, 'package.json'))) {
+      console.warn(
+        yellow(
+          `${bold('Warning:')} could not find the ${bold(
+            'package.json',
+          )} file in the inferred project root, it will be created by your package manager on the appropriate level if necessary.`,
```

Comment:
```suggestion
          `${bold('Warning:')} Prisma could not find a ${bold(
            'package.json',
          )} file in the inferred project root. During the next step, when an auto-install of Prisma package(s) will be attempted, it will then be created by your package manager on the appropriate level if necessary.`,
```